### ISSUE TITLE
Allow to pass Ice config

### ISSUE THREAD

**Issue by jburel** [ome/omero-insight#190]
Allow support for ice config.
This will require change to the gateway to pass handle more info to "login credentials"

cc @dominikl

**Comment by dominikl** [comment]
Does it need a change? If you use `LoginCredentials(String[] args)` constructor you can pass it through. The problem is that the `omero.client` itself has only these two options: Either you log in with username, password, etc. (then you can't  pass through any extra options) or you use the `client(String[] args)` constructor.

**Comment by jburel** [comment]
I should have written "might require"
since I have not look at it in details

**Comment by dominikl** [comment]
By always constructing the `String [] args` in the `LoginCredentials` it would be possible to add a convenience method to the gateway to make it easier to pass through addtional parameters. I'll add it to my draft PR https://github.com/ome/omero-insight/pull/182 .

**Comment by joshmoore** [comment]
For what it's worth, `Ice.Communicator` (used in `omero.client`) should also read the contents of any `$ICE_CONFIG` file automatically, but at the moment, that looks to be disallowed by insight.

**Comment by jburel** [comment]
insight reads the container.xml as the first argument if specified
so the logic needs to be adjusted

**Comment by joshmoore** [comment]
I'm not sure what you're saying. That should be independent of an environment variable, no?

**Comment by jburel** [comment]
the idea of the user was to do ``insight.exe ice.config ``, that will not work since insight will expect in that case ``insight.exe yourowncontainer.xml ``

**Comment by joshmoore** [comment]
Right, but as a workaround perhaps `ICE_CONFIG=ice.config insight.exe` ? I was just hoping that would be easier.

**Comment by dominikl** [comment]
Just adding a reference to the respective forum post: [Set Proxy in BlitzGateway](https://forum.image.sc/t/set-proxy-in-blitzgateway/43780/10)

**Comment by jburel** [comment]
https://github.com/ome/omero-insight/pull/197 partially fixes the issue.
``ICE_CONFIG=pathToIceConfig`` is not supported yet
